src/EnergyPlus/SolarShading.cc

```diff
@@ -363,6 +365,32 @@ void InitSolarCalculations(EnergyPlusData &state)
     state.dataSolarShading->firstTime = false;
 }
 
+void checkShadingSurfaceSchedules(EnergyPlusData &state)
+{
+    // Shading surfaces with a transmittance schedule that is always 1.0 are marked IsTransparent during shading surface input processing
+    // Now that EMS (and other types) actuators are set up, check to see if the schedule has an actuator and reset if needed
+    for (int surfNum = state.dataSurface->ShadingSurfaceFirst; surfNum <= state.dataSurface->ShadingSurfaceLast; ++surfNum) {
+        auto &thisSurface = state.dataSurface->Surface(surfNum);
+        if (!thisSurface.IsTransparent) continue;
+        if (state.dataGlobal->AnyEnergyManagementSystemInModel &&
+            (EMSManager::isScheduleManaged(state, thisSurface.SchedShadowSurfIndex) || state.dataGlobal->anyPluginsOrCallbacksInModel)) {
+            // Transmittance schedule defintely has an actuator or may have one via python plugin or API
+            // Set not transparent so it won't be skipped during shading calcs
```

Comment:
Is it wise to include anyPluginsOrCallbacksInModel if you don't know if python or API acts on this surface?

Reply:
There's no way to know ahead of time if it will be acted on via Python or API, but if we don't assume that, then any code there will have zero impact. If the schedule is not acted on, the only extra cost is a `LookupScheduleValue` during the shading calcs. If the schedule value is 1.0, then the surface gets skipped.

Reply:
This response covers my hesitation.